crates/math/src/multilinear_extension.rs

```diff
@@ -431,6 +477,130 @@ mod tests {
 		assert_eq!(result1, result2);
 	}
 
+	fn get_bits<P, PE>(values: &[PE], start_index: usize) -> Vec<PE::Scalar>
+	where
+		P: PackedField,
+		PE: PackedField<Scalar: ExtensionField<P::Scalar>>,
+	{
+		let new_vals = values
+			.iter()
+			.flat_map(|v| {
+				(P::WIDTH * start_index..P::WIDTH * (start_index + 1))
+					.map(|i| v.get(i))
+					.collect::<Vec<_>>()
+			})
+			.collect::<Vec<_>>();
+
+		new_vals
+	}
+
+	#[test]
+	fn test_evaluate_middle_32b_to_16b() {
+		let mut rng = StdRng::seed_from_u64(0);
+		let values: Vec<_> = repeat_with(|| PackedBinaryField32x1b::random(&mut rng))
```

Comment:
Please use turbofish syntax `.collect::<Vec<_>>()` instead of specifying the variable type.